src/Api/Upload.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Kirby\Api;
+
+use Kirby\Cms\App;
+use Kirby\Filesystem\Dir;
+use Kirby\Filesystem\F;
+use Kirby\Toolkit\Str;
+
+/**
+ * The Upload class supports file uploads in the
+ * context of the API
+ *
+ * @package   Kirby Api
+ * @author    Nico Hoffmann <nico@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier
+ * @license   https://getkirby.com/license
+ * @since     4.3.0
+ */
+class Upload
+{
+	/**
+	 * Handle chunked uploads by merging all chunks
+	 * in the tmp directory and only returning the new
+	 * $source path to the tmp file once complete
+	 */
+	public static function chunk(
+		Api $api,
+		string $source,
+		string $filename
+	): string|null {
+		// if the file is uploaded in chunks
+		if ($isChunk = $api->requestHeaders('X-Upload-Chunk')) {
+			// ensure the directory exists
+			Dir::make($dir = static::tmp());
+
+			// append chunk to tmp file with unique filename (id suffix)
+			$id     = $api->requestHeaders('X-Unique-Upload-Id');
+			$chunk  = fopen($source, 'r');
+			$source = $dir . '/' . $filename;
+			$tmp    = $source . '.' . $id;
+			$part   = fopen($tmp, 'a');
+			stream_copy_to_stream($chunk, $part);
+
+			// if not the last chunk, return early
+			if ($isChunk !== 'last') {
+				return null;
+			}
+
+			// remove id from tmp filename now the file is complete
+			rename($tmp, $source);
+		}
+
+		return $source;
+	}
+
+	/**
+	 * Clean up tmp directory of stale files
+	 */
+	public static function clear(): void
```

Comment:
Shouldn't this rather be `clean`? With `clear` I'd think that the folder is completely emptied.